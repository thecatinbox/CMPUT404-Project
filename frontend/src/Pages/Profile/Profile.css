.profile {
    width: 60%;
    margin: auto;
}

.profile .profile-data {
    text-align: center;
    padding-top: 5%;
}

.profile img {
    width: 20%;
    height: auto;
}