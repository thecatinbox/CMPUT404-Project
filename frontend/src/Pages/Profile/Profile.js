import Post from '../../Components/Post/Post'; 
import React, {useState, useEffect} from 'react';
import './Profile.css';
import TopBar from "../../Components/TopBar/TopBar";

import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogTitle from '@mui/material/DialogTitle';

function Profile() {

  const [open, setOpen] = React.useState(false);
  const [postList, setPostList] = useState([]);
  const [userData, setUserData] = useState([]);

  // const ENDPOINT = 'http://127.0.0.1:8000/server/authors/7dce957d-4ba2-4021-a76a-3ed8c4a06c97/posts/'
  const uuid = localStorage.getItem('uuid'); 
  // console.log(uuid); 
  const POSTS_ENDPOINT = 'http://127.0.0.1:8000/server/authors/' + uuid + '/posts/'; 
  const USER_ENDPOINT = 'http://localhost:8000/server/authors/' + uuid + '/'; 

  useEffect(() => { 
    fetch(POSTS_ENDPOINT, {
      headers: { "Accept": "application/json" },
      method: "GET"
    }).then(response => response.json()).then(postData => {
      setPostList(postData.items);
      // console.log(postData); 
    }); 

    fetch(USER_ENDPOINT , {
      headers: { "Accept": "application/json" },
      method: "GET"
    }).then(response => response.json()).then(userData => {
      setUserData(userData.items);
    }); 
  })

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <>
      <TopBar id="profile"/>
      <div className="profile">
        <div className='profile-data'>
          <img src={userData.profileImage} alt="Profile Image"></img>
          <h2>{userData.displayName}</h2>
          <button onClick={handleClickOpen}>Edit Profile</button>
          <Dialog open={open} onClose={handleClose}>
            <DialogTitle>Edit User Profile</DialogTitle>
            <DialogContent>
              <TextField margin="dense" id="displayName}" label="Display Name" value={userData.displayName} variant="standard" fullWidth/>
              <TextField margin="dense" id="github" label="GitHub URL" value={userData.github} variant="standard" fullWidth/>
              <TextField margin="dense" id="profileImage" label="Profile Image URL" value={userData.profileImage} variant="standard" fullWidth/>
            </DialogContent>
            <DialogActions>
              <Button onClick={handleClose}>Cancel</Button>
              <Button onClick={handleClose}>Save</Button>
            </DialogActions>
          </Dialog>
        </div>

        <div className='post-data'>
          {postList.map(function(post){
              return <Post post={post} key={post.id}/>;
          })}
        </div>
      </div>
    </>
  );
}

export default Profile;