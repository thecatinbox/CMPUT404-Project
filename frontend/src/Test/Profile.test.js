import Profile from "../Pages/Profile/Profile";
import React from 'react';
import renderer from 'react-test-renderer';

it('profile page renders correctly', () => {
    const component = renderer.create(
      <Profile />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("edit button can be clicked ", () => {
  const component = renderer.create(
    <Profile />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


/* 
it('exists a Edit Profile button', ()=>{
  const wrapper = render(<Button>Edit Profile</Button>).queryByText('Edit Profile')
  expect(wrapper).toBeTruthy()
})
it('exists a Cancel button', ()=>{
  const wrapper = render(<Button>Cancel</Button>).queryByText('Cancel')
  expect(wrapper).toBeTruthy()
})
it('exists a Save button', ()=>{
  const wrapper = render(<Button>Save</Button>).queryByText('Cancel')
  expect(wrapper).toBeTruthy()
})
*/ 