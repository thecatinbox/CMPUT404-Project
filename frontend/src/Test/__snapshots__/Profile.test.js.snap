// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit button can be clicked  1`] = `
Array [
  <div
    className="topnav"
  >
    <a
      href="/home"
      id="home"
    >
      Home
    </a>
    <a
      href="/inbox"
      id="inbox"
    >
      Inbox
    </a>
    <a
      href="/friends"
      id="friends"
    >
      Friends
    </a>
    <a
      href="/profile"
      id="profile"
    >
      Profile
    </a>
    <a
      href="/signin"
      id="logout"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-right-from-bracket "
        data-icon="right-from-bracket"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <div
      className="search-container"
    >
      <form
        action="/search"
      >
        <input
          name="username"
          placeholder="Search.."
          type="text"
        />
        <button
          type="submit"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-magnifying-glass "
            data-icon="magnifying-glass"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </form>
    </div>
  </div>,
  <div
    className="profile"
  >
    <div
      className="profile-data"
    >
      <img
        alt="Profile Image"
        src="https://i.imgur.com/k7XVwpB.jpeg"
      />
      <h2 />
      <button
        onClick={[Function]}
      >
        Edit Profile
      </button>
    </div>
    <div
      className="post-data"
    />
  </div>,
]
`;

exports[`profile page renders correctly 1`] = `
Array [
  <div
    className="topnav"
  >
    <a
      href="/home"
      id="home"
    >
      Home
    </a>
    <a
      href="/inbox"
      id="inbox"
    >
      Inbox
    </a>
    <a
      href="/friends"
      id="friends"
    >
      Friends
    </a>
    <a
      href="/profile"
      id="profile"
    >
      Profile
    </a>
    <a
      href="/signin"
      id="logout"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-right-from-bracket "
        data-icon="right-from-bracket"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <div
      className="search-container"
    >
      <form
        action="/search"
      >
        <input
          name="username"
          placeholder="Search.."
          type="text"
        />
        <button
          type="submit"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-magnifying-glass "
            data-icon="magnifying-glass"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </form>
    </div>
  </div>,
  <div
    className="profile"
  >
    <div
      className="profile-data"
    >
      <img
        alt="Profile Image"
        src="https://i.imgur.com/k7XVwpB.jpeg"
      />
      <h2 />
      <button
        onClick={[Function]}
      >
        Edit Profile
      </button>
    </div>
    <div
      className="post-data"
    />
  </div>,
]
`;
